File: 0846-hand-of-straights/0846-hand-of-straights.cpp
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int k) {
        
        
        int n=hand.size();if(n%k!=0)return false;
        
        map<int,int>mp;
        for(int i=0;i<n;i++)mp[hand[i]]++;
        
        int m=n/k;
        
        for(int i=1;i<=m;++i)
        {
            auto it=mp.begin();
            
            int val=it->first;
            
            for(int j=val;j<val+k;j++)
            {
                if(mp.find(j)==mp.end())return false;
                
                 mp[j]--;
                
                if(mp[j]==0)
                {
                    mp.erase(j);
                }
            }
        }
        return true;
    }
};

File: 0846-hand-of-straights/0846-hand-of-straights.py
class Solution:
    def isNStraightHand(self, hand: List[int], k: int) -> bool:
        
        n=len(hand)
        
        if n % k != 0:
            return False
    
        
        m=n//k
        
        mp={}
        
        for idx in hand:
            if idx in mp:
                mp[idx]+=1
            else:
                mp[idx]=1
                
        for i in range(1,m+1):
            
            val=min(mp)
            
            for j in range(val,val+k):
                
                if j not in mp:
                    return False
                
                mp[j]-=1
                
                if mp[j]==0:
                    del mp[j]
                    
        return True         
                    
                      
            
            
            